### PR TITLE
Add Cppdbg Tests to MIDebugEngine.sln

### PR DESCRIPTION
**Pull request by WardenGnaw**
This PR migrates over DebuggerTesting, CppTests, and DebugAdapterRunner to the MIEngine repository. 

- Debugger Testing is the infrastructure used to build c++ debuggee projects and let tests know which debugger to run. 
- CppTests holds the debuggees, configurations, and tests to run aganist.
- DebugAdapterRunner is a tool to help compare Request/Responses for the DebugAdapterProtocol (DAP).

These test projects are added under MIEngine_Root/test and are added to the MIDebugEngine.sln.
It will output it in `$(MIEngineRoot)bin\DebugAdapterProtocolTests\$(Configuration)`

Aside: Added the build props to the MIDebugEngine.sln too. 

TODO: Handling CrossPlatform builds, updating the releases, and running tests.

### CODE REVIEW

File: test/DebugAdapterRunner/DebugAdapterRunner.nuspec [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
```
Comment (gregg-miskelly) [inline test/DebugAdapterRunner/DebugAdapterRunner.nuspec:2]: >package [](http://example.com/codeflow?start=1&length=7)

Is this still used?

File: test/DebugAdapterRunner/DebugAdapterRunner.csproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    <PackageReference Include="Newtonsoft.Json" Version="$(Newtonsoft_Json_VSCode_Version)" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <DropSignedFile Include="$(OutputPath)dar.dll" />
+    <DropUnsignedFile Include="$(OutputPath)dar.pdb" />
+  </ItemGroup>
+
+  <Import Project="..\..\build\DropFiles.targets" />
```
Comment (gregg-miskelly) [inline test/DebugAdapterRunner/DebugAdapterRunner.csproj:23]: >Import [](http://example.com/codeflow?start=3&length=6)

The SourceLink support in MIEngine is in miengine.targets. Seems like somehow we should include that.

File: test/CppTests/CppTests.csproj [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  </Target>
+
+  <ItemGroup>
+    <DropSignedFile Include="$(OutputPath)CppTests.dll" />
+    <DropUnsignedFile Include="$(OutputPath)CppTests.pdb" />
+    <DropSignedFile Include="$(OutputPath)DebuggerTesting.dll" />
+    <DropUnsignedFile Include="$(OutputPath)DebuggerTesting.pdb" />
+  </ItemGroup>
+
+  <Import Project="..\..\build\DropFiles.targets" />
```
Comment (gregg-miskelly) [inline test/CppTests/CppTests.csproj:48]: >Import [](http://example.com/codeflow?start=3&length=6)

Same comments for this project: do we want to use the nuspec, and we should include SourceLink.

**Review by gregg-miskelly — APPROVED** [review]
Otherwise, I didn't review everything, but it seems good to me.

File: test/CppTests/CppTests.nuspec [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    <iconUrl>https://aka.ms/vsextensibilityicon</iconUrl>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <description>Contains C++ Tests and Debuggees used for testing C++ Debug Adapters. This package should only be used for testing owned by the Microsoft Visual Studio Diagnostics team and isn't supported for other purposes.</description>
+    <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+  </metadata>
+  <files>
+    <file src="$binRoot$CppTests.dll" target="lib\net5.0" />
+    <file src="$binRoot$CppTests.pdb" target="lib\net5.0" />
+    <file src="$binRoot$DebuggerTesting.dll" target="lib\net5.0" />
+    <file src="$binRoot$DebuggerTesting.pdb" target="lib\net5.0" />
```
Comment (gregg-miskelly) [inline test/CppTests/CppTests.nuspec:20]: >$binRoot$DebuggerTesting [](http://example.com/codeflow?start=15&length=24)

Oh, I guess the same for DebuggerTesting too

File: test/DebugAdapterRunner/DebugAdapterRunner.nuspec (hunk 1 above)
Comment (WardenGnaw) [reply test/DebugAdapterRunner/DebugAdapterRunner.nuspec:2]: Yep. We will still need to ship DAR independently for VsDbg-UITests. 

I'll be adding in a yaml in a later PR to release DAR from this repo.

**Review by gregg-miskelly — APPROVED** [review]

**Pull request merged**